src/os/bluestore/BlueStore.h

```diff
@@ -514,15 +532,46 @@ class BlueStore : public ObjectStore,
 //#define CACHE_BLOB_BL  // not sure if this is a win yet or not... :/
 
   /// in-memory blob metadata and associated cached buffers (if any)
+  struct Blob;
+  typedef boost::intrusive_ptr<Blob> BlobRef;
+
   struct Blob {
     MEMPOOL_CLASS_HELPERS();
 
     std::atomic_int nref = {0};     ///< reference count
     int16_t id = -1;                ///< id, for spanning blobs only, >= 0
     int16_t last_encoded_id = -1;   ///< (ephemeral) used during encoding only
-    SharedBlobRef shared_blob;      ///< shared blob state (if any)
+
+    const BufferSpaceBase* buffer_space() const {
+      if (unlikely(is_shared())) {
```

Comment:
my understanding from the linux kernel is that using likely/unlikely can have a high penalty if you're wrong, so it should really only be for exceptional cases.  i think cloned blobs aren't so uncommon?

Reply:
yeah, probably you're right. will remove.